### DISCUSSION TITLE
strange behavior of res.writeHeader()

### DISCUSSION THREAD

**Discussion by kng83** [uNetworking/uWebSockets.js#121] [Q&A] [answered]
I have code like below:
.get('id', (res,req)=>{
  res.writeHeader('Content-Type','text/html; charset=utf-8');
  res.writeHeader('Content-Type','text/json');
  res.end('');
})
When I send request to localhost:9001/id  I get response from server with 2 keys with the same name 'Content-Type' (look on Response Header). It seems like writeHeader() doesn't write the same key but push every key-value pair  like new one.
       It will be nice to  put function like res.removeHeader() from express because it is not possible to remove response Header like uWebSockets: v0.15. I'd like to  remove it  for safety reasons. I don't want to, that someone knows, what server do I use.

I use windows 7
node 10.15.3
"uWebSockets.js": "github:uNetworking/uWebSockets.js#v15.8.0"

**Comment by ghost** [comment]
Well you have to make up your mind before you commit ;)

**Comment by kng83** [comment]
You right.But someone has told me once :It makes sense to follow the docs.  Write is not push.

**Comment by ghost** [comment, marked as answer]
https://unetworking.github.io/uWebSockets.js/generated/interfaces/httpresponse.html#writeheader

Writes key and value to HTTP response.

This is what it does. It writes (in order) what you hand it. That's why lazily written "middlewares" won't fit here - you have to know what you want to write before you write it, well at least you cannot overwrite

**Comment by coder456** [comment]
Is it necessary to force the `uwebsockets` header, it would be preferable to not broadcast the software and version running on the server
